### ISSUE TITLE
About LUNA16 Dataset Preprocess

### ISSUE THREAD

**Issue by kimm51** [MIC-DKFZ/nnDetection#148]
Hello,

I  redrawed mydataset as a LUNA16 dataset  in order to run it in nnDetection.  I firstly preprared  my dataset without any error. However, When I implemented preprocess.py , faced an error like in this screenshot.
![Screenshot from 2023-02-21 15-24-02](https://user-images.githubusercontent.com/125980203/220344392-18bdf377-ac4e-4026-bba2-47cf5080db3e.png)

Is there any recommendation to overcome this?

Thanks,

**Comment by mibaumgartner** [comment]
The error indicates that there are no bounding boxes in the dataset, can you make sure that the labels exist and there are objects in the labels?

**Comment by kimm51** [comment]
Hello again,

First of all, thank you for your return. I have got hdf5 dataset and its pixel coordinate annotated boxes. Firstly, I transformed my pixel annotations  to world coordinate annotations like ((center_x(x+width/2)    center_y(y+height/2) z(slice_number) 2*r(mean([width,height]))  )) and saved it annotations.csv. (with id) . For mhd file, I  determined a scheme like (attached)this
ObjectType = Image
NDims = 3
BinaryData = True
BinaryDataByteOrderMSB = False
CompressedData = False
TransformMatrix = 1 0 0 0 1 0 0 0 1
Offset = 0 0 0
CenterOfRotation = 0 0 0
AnatomicalOrientation = RAI
ElementSpacing = 1 1 1
DimSize = 320 320 36
ElementType = MET_SHORT
ElementDataFile = 1000001.raw

 and just only changed offset, element spacing and  voxel size.  For raw dataset, I read each slice and converted it range 0-255 and then amplified 10000 to save int16.  All dataset divided 10 folders(containing raw and mhd file) with annotations csv. I have prepared dataset but in preprocess section (especially creating D3V001_3d) I have faced  this problem both in docker and external running. Do I skip something here?

Best,

**Comment by mibaumgartner** [comment]
It feels like the current approach is rather complicated.

1) The data shouldn't be preprocessed before nnDetection, so if you have access to the raw CT images, please use them directly.

2) Also you are effectively loosing information by changing your bounding box a circle as in LUNA, I would recommend writing a small script which fits an ellipsoid into the bounding box, that would preserve the original shape of the box instead of reducing it to a circle.

Since you are running a custom preparation of the data, I would recommend checking the labels manually. I'm happy to assist with information regarding the format etc. but it is really hard for me to exclude any other potential pitfalls of the other preparation steps.

**Comment by kimm51** [comment]
Hello,

My data is not CT, it is MRI. Consequently, there is no raw data and I decided to convert my data to raw data. 

1. I must preprocess. Because it contains kspace and its rss.
2.  I tested it over image. (attached  a slice here). After that, I converted all annotations to this world coordinate format. So that, x y wid height converted  center_x center_y slice_number(annotated) and r.
 ![image](https://user-images.githubusercontent.com/125980203/220906848-b74bb428-3955-4da2-9079-532e0fd69d22.png)

Thank you!
